ghstack/config.py

```diff
@@ -131,8 +136,9 @@ def read_config(*, request_circle_token: bool = False) -> Config:  # noqa: C901
         default_project_dir = 'fbcode/caffe2'
 
     if write_back:
-        config.write(open(os.path.expanduser('~/.ghstackrc'), 'w'))
-        logging.info("NB: configuration saved to ~/.ghstackrc")
+        config_path = '~/.ghstackrc' if config_path is None else config_path
+        config.write(open(os.path.expanduser(config_path), 'w'))
```

Comment:
It seems like it would be better if you just setup the invariant that `config_path` is THE CONFIG PATH and then you can just unconditionally write to it.

Reply:
what if I located another config in a subfolder? But if it misses some fields. Then I would write there. Really core case though. I mainly care about config either existing or not existing at all :) But the code so far was enriching existing config, so I added support for it as well.

Reply:
The logic for enriching is because the way this code evolved, I initially didn't put in enough config entries, and then later when I added a new config I didn't want to clobber the old code.

In general I'd expect it to write to the "closest" config file it finds, or if it doesn't find any, to a default location (`~/.ghstackrc` is what I'd expect.)

Reply:
I hope we don't hit a race by users doing several writes to config files by sending several RPs in parallel, lol